lib/sentry/opentelemetry/span_processor.ex

```diff
@@ -0,0 +1,199 @@
+if Code.ensure_loaded?(OpenTelemetry) do
+  defmodule Sentry.OpenTelemetry.SpanProcessor do
+    @moduledoc false
+
+    require OpenTelemetry.SemConv.ClientAttributes, as: ClientAttributes
+    require OpenTelemetry.SemConv.Incubating.DBAttributes, as: DBAttributes
+    require OpenTelemetry.SemConv.Incubating.HTTPAttributes, as: HTTPAttributes
+    require OpenTelemetry.SemConv.Incubating.URLAttributes, as: URLAttributes
+    require OpenTelemetry.SemConv.Incubating.MessagingAttributes, as: MessagingAttributes
+    @behaviour :otel_span_processor
+
+    require Logger
+
+    alias Sentry.{Transaction, OpenTelemetry.SpanStorage, OpenTelemetry.SpanRecord}
+    alias Sentry.Interfaces.Span
+
+    @impl true
+    def on_start(_ctx, otel_span, _config) do
+      span_record = SpanRecord.new(otel_span)
+
+      SpanStorage.store_span(span_record)
+
+      otel_span
+    end
+
+    @impl true
+    def on_end(otel_span, _config) do
+      span_record = SpanRecord.new(otel_span)
+
+      SpanStorage.update_span(span_record)
+
+      if span_record.parent_span_id == nil do
+        root_span_record = SpanStorage.get_root_span(span_record.span_id)
+        child_span_records = SpanStorage.get_child_spans(span_record.span_id)
+        transaction = build_transaction(root_span_record, child_span_records)
+
+        result =
+          case Sentry.send_transaction(transaction) do
```

Comment:
Maybe we can do this in a later PR, but I’m pretty scared of sending an HTTP request *every time a span ends*. Don't implementations usually buffer these?

Reply:
@whatyouhide I actually hit rate limits in my app on prod because of this so it's definitely something we want to address. I'll do a follow-up PR after this gets merged so that we're not lost in the comments and commits.